### PR TITLE
Make module context-aware

### PR DESCRIPTION
**Pull request by rzhao271**
Fixes microsoft/vscode#120570

- Added tests
- Set Appveyor to only use Node 12 and 14
- Updated some dependencies (we're still using tslint, though)

### CODE REVIEW

**Review by deepak1556 — CHANGES_REQUESTED** [review]
Please add tests that uses the module simultaneously from main thread as well as nodejs worker threads

File: appveyor.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ environment:
     - nodejs_version: "10"
     - nodejs_version: "11"
```
Comment (Tyriar) [inline appveyor.yml:8]: Let's remote 10 and 11, they're pretty old at this point.

**Review by Tyriar — APPROVED** [review]

**Review by bpasero — APPROVED** [review]
👏  for the test

**Comment by deepak1556** [comment]
@Tyriar appveyor is not running on this repo due to permission issues https://ci.appveyor.com/project/Tyriar/vscode-windows-process-tree/builds/38563417, can you reauthorize. Thanks!

**Comment by Tyriar** [comment]
Fixed it

**Comment by rzhao271** [comment]
I'll go change the tests because the current worker tests don't call into native code.

**Comment by deepak1556** [comment]
Additionally can you add a test that checks the memory usage of the process, use `getProcessList` after multiple workers have exited to ensure there is no memory leak.

**Review by Tyriar — CHANGES_REQUESTED** [review]
Let's delete all the nodejs versions from appveyor.yml that don't support worker_threads

**Comment by rzhao271** [comment]
I didn't add the memory leak test because the delta between the initial memory and final memory amounts, even after enabling GC (by passing `--enable-gc` to mocha in `package.json` and by using `global.gc()` in the memory leak test), kept varying too much.

**Comment by rzhao271** [comment]
Ready for review again

**Review by deepak1556 — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]
Thanks, very thorough 🙂

**Pull request merged**